packages/dx-react-grid-demos/src/demo-sources/grid-export/advanced-customization.partial.jsxt

```diff
@@ -0,0 +1,213 @@
+);
+
+// worksheet customization
+/* eslint-disable no-param-reassign */
+const customizeCell = (cell, row, column) => {
+  if (row.groupedBy && groupColors[row.groupedBy]) {
+    cell.fill = { type: 'pattern', pattern: 'solid', fgColor: { argb: groupColors[row.groupedBy] } };
+    cell.font = { color: { argb: '000000' } };
+  }
+  if (row.OrderDate < new Date(2014, 2, 3)) {
+    cell.font = { color: { argb: 'AAAAAA' } };
+  }
+  if (row.SaleAmount > 15000) {
+    if (column.name === 'SaleAmount') {
+      cell.font = { color: { argb: '000000' } };
+      cell.fill = { type: 'pattern', pattern: 'solid', fgColor: { argb: 'FFBB00' } };
+    }
+  }
+  if (column.name === 'SaleAmount') {
+    cell.numFmt = '$0';
+  }
+};
+
+const customizeSummaryCell = (cell /* col, summary */) => {
+  cell.font = { italic: true };
+};
+
+const customizeHeader = (worksheet) => {
+  const generalStyles = {
+    font: { bold: true },
+    fill: {
+      type: 'pattern', pattern: 'solid', fgColor: { argb: 'D3D3D3' }, bgColor: { argb: 'D3D3D3' },
+    },
+    alignment: { horizontal: 'left' },
+  };
+  for (let rowIndex = 1; rowIndex < 6; rowIndex += 1) {
+    worksheet.mergeCells(rowIndex, 1, rowIndex, 2);
+    worksheet.mergeCells(rowIndex, 3, rowIndex, 4);
+    Object.assign(worksheet.getRow(rowIndex).getCell(1), generalStyles);
+    Object.assign(worksheet.getRow(rowIndex).getCell(3), generalStyles);
+  }
+  worksheet.getRow(1).height = 20;
+  worksheet.getRow(1).getCell(1).font = { bold: true, size: 16 };
+  worksheet.getRow(1).getCell(3).numFmt = 'd mmmm yyyy';
+  worksheet.getRow(1).getCell(3).font = { bold: true, size: 16 };
+  worksheet.getColumn(1).values = ['Sale Amounts:', 'Company Name:', 'Address:', 'Phone:', 'Website:'];
+  worksheet.getColumn(3).values = [new Date(), 'K&S Music', '1000 Nicllet Mall Minneapolis Minnesota', '(612) 304-6073', 'www.nowebsitemusic.com'];
+};
+
+const customizeFooter = (worksheet) => {
+  const { lastRow } = worksheet;
+  let currentRowIndex = lastRow.number + 2;
+  for (let rowIndex = 0; rowIndex < 3; rowIndex += 1) {
+    worksheet.mergeCells(currentRowIndex + rowIndex, 1, currentRowIndex + rowIndex, 4);
+    Object.assign(worksheet.getRow(currentRowIndex + rowIndex).getCell(1), { font: { bold: true }, alignment: { horizontal: 'right' } });
+  }
+  worksheet.getRow(currentRowIndex).getCell(1).value = 'If you have any questions, please contact John Smith.';
+  currentRowIndex += 1;
+  worksheet.getRow(currentRowIndex).getCell(1).value = 'Phone: +111-111';
+  currentRowIndex += 1;
+  worksheet.getRow(currentRowIndex).getCell(1).value = 'For demonstration purposes only';
+  worksheet.getRow(currentRowIndex).getCell(1).font = { italic: true };
+};
+/* eslint-enable no-param-reassign */
+
+const onSave = (workbook) => {
+  workbook.xlsx.writeBuffer().then((buffer) => {
+    saveAs(new Blob([buffer], { type: 'application/octet-stream' }), 'DataGrid.xlsx');
+  });
+};
```

Comment:
Can we avoid using third-party libraries for save file? I think we should illustrate it because our users should not download some additionals.
But we should say about this easy way, btw

Reply:
It's a pretty complicated task:
https://github.com/eligrey/FileSaver.js/blob/master/src/FileSaver.js

Reply:
OMG 😱